python/pysmurf/client/util/SmurfFileReader.py

```diff
@@ -0,0 +1,254 @@
+                                                    # 4 bytes, unused, 4x
+                           'external_time'       ,  # 5 bytes, uint64, Q (3 extra bytes)
+                           'control_field'       ,  # 1 byte, B
+                           'test_params'         ,  # 1 byte, B
+                                                    # 6 bytes, unused, 6x
+                           'num_rows'            ,  # 2 bytes, uint16, H
+                           'num_rows_reported'   ,  # 2 bytes, uint16, H
+                                                    # 4 bytes, unused, 4x
+                           'row_length'          ,  # 2 bytes, uint16, H
+                           'data_rate'              # 2 bytes, uint16, H
+                                                    # 4 bytes, unused, 4x
+                         ] )
+
+# Default header as a named tuple
+RogueHeader = namedtuple( 'RogueHeader',
+                         [ 'size'                ,  # 4 Bytes, uint32, I
+                           'flags'               ,  # 2 bytes, uint16, H
+                           'error'               ,  # 1 bytes, uint8,  B
+                           'channel'                # 1 bytes, uint8,  B
+                         ] )
+
+
+
+class SmurfStreamReader(object):
+
+    def __init__(self, files, *, isRogue=True, metaEnable=False, chanCount=None):
+        self._isRogue    = isRogue
+        self._metaEnable = metaEnable
+        self._chanCount  = chanCount
+        self._currFile   = None
+        self._fileSize   = 0
+        self._header     = None
+        self._data       = None
+        self._config     = {}
+        self._currCount  = 0
+        self._totCount   = 0
+
+        if isinstance(files,list):
+            self._fileList = files
+        else:
+            self._fileList = [files]
+
+        # Check to make sure all the files are readable
+        for fn in self._fileList:
+            if not os.access(fn,os.R_OK):
+                raise Exception(f"Unable to read file {fn}")
+
+
+    def _nextRecord(self):
+        """
+        Process next record, return true on success
+        """
+        recEnd = 0
+
+        # We are at the end of the file
+        if self._currFile.tell() == self._fileSize:
+            return False
+
+        # Use Rogue format
+        if self._isRogue:
+
+            # keep reading until we get a data channel
+            while True:
+
+                # Hit end of file
+                if self._currFile.tell() == self._fileSize:
+                    return False
+
+                # Not enough data left in the file
+                if (self._fileSize - self._currFile.tell()) < RogueHeaderSize:
+                    printf(f"Waring: File under run reading {self._currFName}")
+                    return False
+
+                # Read in Rogue header data
+                rogueHeader = RogueHeader._make(struct.Struct(RogueHeaderPack).unpack(self._currFile.read(RogueHeaderSize)))
```

Comment:
@slacrherbst Did you verify if this line unpack the rogue header correctly? I tried doing it like this, but it was not unpacking the second 32-bit value correctly. So, I had to unpack the rogue header like this: 

https://github.com/slaclab/pysmurf/blob/355067995c91640fe15e783d2a62b81794dbb82c/python/pysmurf/client/util/smurf_util.py#L841-L867

Reply:
I am working on a core rogue version of this. I will test it there and update this PR if required.

Reply:
The rogue header unpack is working for me in testing.